be/src/storage/tablet_updates.cpp

```diff
@@ -233,33 +233,67 @@ Status TabletUpdates::_load_from_pb(const TabletUpdatesPB& tablet_updates_pb) {
 
     // Load delete vectors and update RowsetStats.
     // TODO: save num_dels in rowset meta.
+
+    std::map<uint32_t, std::vector<std::pair<int32_t, DelVectorPtr>>> del_vector_by_rsid;
+    uint32_t max_rsid = 0;
+    uint32_t min_rsid = 1 << 31;
+    for (auto& [rsid, rowset] : _rowsets) {
+        if (unapplied_rowsets.find(rsid) == unapplied_rowsets.end()) {
+            std::vector<std::pair<int32_t, DelVectorPtr>> del_vectors;
+            del_vectors.resize(rowset->num_segments());
+            del_vector_by_rsid[rsid] = del_vectors;
+            if (max_rsid <= rsid) {
+                max_rsid = rsid + rowset->num_segments();
+            }
+            if (min_rsid >= rsid) {
+                min_rsid = rsid;
+            }
+        }
+    }
+
+    RETURN_IF_ERROR(TabletMetaManager::del_vector_iterate(
+            _tablet.data_dir()->get_meta(), _tablet.tablet_id(), min_rsid, max_rsid,
+            [&](uint32_t segment_id, int64_t version, std::string_view value) -> bool {
+                auto iter = del_vector_by_rsid.lower_bound(segment_id);
+                if (iter == del_vector_by_rsid.end()) {
+                    std::string msg = strings::Substitute("found delete vector of non exist rowset");
+                    _set_error(msg);
+                    LOG(ERROR) << msg;
+                    return false;
+                }
+                uint32_t rsid = iter->first;
+                uint32_t idx = segment_id - rsid;
+                DCHECK(idx < iter->second.size());
+                if (version >= iter->second[idx].first) {
```

Comment:
what's the initial value of `iter->second[idx].first` ? is it 0 for sure?

Reply:
> what's the initial value of `iter->second[idx].first` ? is it 0 for sure?

Has initialized as 0.